app/routes.py

```diff
@@ -1 +1,227 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+from sqlalchemy import asc, desc 
+import datetime
+import requests
+import os 
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST"])
+def create_tasks():
+    request_body = request.get_json()
+    new_task = Task.from_dict(request_body) 
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    task_dict = {}
+    task_dict["task"] = {"id":new_task.task_id, "title": new_task.title,"description":new_task.description, "is_complete":bool(new_task.completed_at)}
+
+    return make_response(jsonify(task_dict), 201) 
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query:
+        if "asc" in sort_query: 
+            tasks = Task.query.order_by(Task.title.asc())
+        elif "desc" in sort_query: 
+            tasks = Task.query.order_by(Task.title.desc()) 
+    else: 
+        tasks = Task.query.all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+    return jsonify(tasks_response) 
+    
+
+def validate_model(cls, model_id):
+    try: 
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message": f"{cls.__name__} {model_id} invalid"}, 400)) 
+
+    model = cls.query.get(model_id)
+    
+    if not model:
+        abort(make_response({"message": f"{cls.__name__} {model_id} not found"}, 404)) 
+
+    return model
+
+@tasks_bp.route("/<task_id>", methods = ["GET"])
+def get_one_task(task_id):
+    task = validate_model(Task,task_id)
+
+    task_dict = {}
+    task_dict["task"] = {"id":task.task_id, "title": task.title,"description":task.description, "is_complete":bool(task.completed_at)}
+
+    if bool(task.goal_id) == True:
+        task_with_goal_dict = {}
+        task_with_goal_dict["task"] = {"id":task.task_id,"goal_id":task.goal_id,"title": task.title,"description":task.description, "is_complete":bool(task.completed_at)}
+        return make_response(jsonify(task_with_goal_dict), 200)
+
+    return make_response(jsonify(task_dict), 200) 
```

Comment:
And as I suspected, once I expanded your implementation of `to_dict` with my above suggestion, I was able to replace _all_ of this code with:

```python
task_dict = {
    "task": task.to_dict()
}

return task_dict
```

Note that I also replaced the call to `make_response` in the `return` statement with just returning `task_dict` because since returning a JSON object with a 200 is so common, Flask does that by default, so there's no need to call `make_response`.

In fact, if all you are changing in your response is the status code, you can even get away with returning a tuple:

```python
return task_dict, 201
```

which is equivalent to 

```python
return make_response(task_dict, 201)
```

Reply:
Anyway, obviously this applies to other parts of this code, but I think the suggestion is plenty clear already, so I won't belabor the point more than I already have.